pkg/scalers/mssql_scaler.go

```diff
@@ -0,0 +1,280 @@
+		return nil, fmt.Errorf("error establishing mssql connection: %s", err)
+	}
+
+	return &mssqlScaler{
+		metadata:   meta,
+		connection: conn,
+	}, nil
+}
+
+// parseMSSQLMetadata takes a ScalerConfig and returns a mssqlMetadata or an error if the config is invalid
+func parseMSSQLMetadata(config *ScalerConfig) (*mssqlMetadata, error) {
+	meta := mssqlMetadata{}
+
+	// Query
+	if val, ok := config.TriggerMetadata["query"]; ok {
+		meta.query = val
+	} else {
+		return nil, fmt.Errorf("no query given")
+	}
+
+	// Target query value
+	if val, ok := config.TriggerMetadata["targetValue"]; ok {
+		targetValue, err := strconv.Atoi(val)
+		if err != nil {
+			return nil, fmt.Errorf("targetValue parsing error %s", err.Error())
+		}
+		meta.targetValue = targetValue
+	} else {
+		return nil, fmt.Errorf("no targetValue given")
+	}
+
+	// Connection string, which can either be provided explicitly or via the helper fields
+	switch {
+	case config.AuthParams["connectionString"] != "":
+		meta.connectionString = config.AuthParams["connectionString"]
+	case config.TriggerMetadata["connectionStringFromEnv"] != "":
+		meta.connectionString = config.ResolvedEnv[config.TriggerMetadata["connectionStringFromEnv"]]
+	default:
+		meta.connectionString = ""
+		if val, ok := config.TriggerMetadata["host"]; ok {
+			meta.host = val
+		} else {
+			return nil, fmt.Errorf("no host given")
+		}
+
+		if val, ok := config.TriggerMetadata["port"]; ok {
+			port, err := strconv.Atoi(val)
+			if err != nil {
+				return nil, fmt.Errorf("port parsing error %s", err.Error())
+			}
+
+			meta.port = port
+		}
+
+		if val, ok := config.TriggerMetadata["username"]; ok {
+			meta.username = val
+		}
+
+		// database is optional in SQL s
+		if val, ok := config.TriggerMetadata["database"]; ok {
+			meta.database = val
+		}
+
+		if config.AuthParams["password"] != "" {
+			meta.password = config.AuthParams["password"]
+		} else if config.TriggerMetadata["passwordFromEnv"] != "" {
+			meta.password = config.ResolvedEnv[config.TriggerMetadata["passwordFromEnv"]]
+		}
+	}
+
+	// get the metricName, which can be explicit or from the (masked) connection string
+	if val, ok := config.TriggerMetadata["metricName"]; ok {
+		meta.metricName = kedautil.NormalizeString(fmt.Sprintf("mssql-%s", val))
+	} else {
+		if meta.database != "" {
+			meta.metricName = kedautil.NormalizeString(fmt.Sprintf("mssql-%s", meta.database))
+		} else if meta.host != "" {
+			meta.metricName = kedautil.NormalizeString(fmt.Sprintf("mssql-%s", meta.host))
+		} else if meta.connectionString != "" {
+			// The mssql provider supports of a variety of connection string formats. Instead of trying to parse
+			// the connection string and mask out sensitive data, play it safe and just hash the whole thing.
+			connectionStringHash := sha256.Sum256([]byte(meta.connectionString))
+			meta.metricName = kedautil.NormalizeString(fmt.Sprintf("mssql-%x", connectionStringHash))
+		} else {
+			meta.metricName = "mssql"
+		}
```

Comment:
i think the static checker wants it to be 

```golang
		switch {
		case meta.database != "":
			meta.metricName = kedautil.NormalizeString(fmt.Sprintf("mssql-%s", meta.database))
		case meta.host != "":
			meta.metricName = kedautil.NormalizeString(fmt.Sprintf("mssql-%s", meta.host))
		case  meta.connectionString != "":
			// The mssql provider supports of a variety of connection string formats. Instead of trying to parse
			// the connection string and mask out sensitive data, play it safe and just hash the whole thing.
			connectionStringHash := sha256.Sum256([]byte(meta.connectionString))
			meta.metricName = kedautil.NormalizeString(fmt.Sprintf("mssql-%x", connectionStringHash))
		default:
			meta.metricName = "mssql"
		}
```